src/chainstate/burn/db/sortdb.rs

```diff
@@ -2661,11 +2664,17 @@ impl SortitionDB {
     }
 
     fn add_indexes(&mut self) -> Result<(), db_error> {
-        let tx = self.tx_begin()?;
-        for row_text in SORTITION_DB_INDEXES {
-            tx.execute_batch(row_text)?;
+        // do we need to instantiate indexes?
+        // only do a transaction if we need to, since this gets called each time the sortition DB
+        // is opened.
+        let exists : i64 = query_row(self.conn(), "SELECT 1 FROM sqlite_master WHERE type = 'index' AND name = ?1", &[LAST_SORTITION_DB_INDEX])?.unwrap_or(0);
+        if exists == 0 {
+            let tx = self.tx_begin()?;
+            for row_text in SORTITION_DB_INDEXES {
+                tx.execute_batch(row_text)?;
+            }
+            tx.commit()?;
```

Comment:
I think the first query should be included in the tx.

Reply:
How often is this invoked? This is a database schema check

Reply:
It's called by `SortitionDB::connect()`, which gets called via `Burnchain::sync_with_indexer()`, which gets called every time you want to download a Bitcoin block.  So, it's pretty often.

Reply:
I see. I guess that's not ideal (Burnchain sync with indexer shouldn't need to reopen a connection on every invocation), but it's probably not too burdensome as long as it doesn't try to initiate a transaction.

This is fine, I suppose, the threat of `exists` being inconsistent with the index creation tx is pretty low, because the index creation is idempotent anyways.